linera-core/src/client/mod.rs

```diff
@@ -1822,7 +1822,7 @@ where
                 chain_state_view
                     .manager
                     .get()
-                    .pending_blobs
+                    .proposed_pending_blobs
                     .get(&blob_id)
                     .cloned()
```

Comment:
We seem to be calling `read_local_blobs` _before_ we call `local_node.handle_block_proposal`; is there a scenario where we expect them to be in there at that point?

Reply:
But won't they only be added to `proposed_pending_blobs` [in this call to `handle_block_proposal`](https://github.com/linera-io/linera-protocol/blob/13d17a6884f3b190f5b81dde0ce889f4f9135ea6/linera-core/src/client/mod.rs#L1931)? It looks like `read_local_blobs` is only called _before_ that, so we don't expect the blobs to be in there?

Reply:
So, the issue is that there might be blobs in the current `proposal`, which are not yet stored on the node, but we might request them via `read_blobs_local`?

Reply:
Yes… at least if it's a proposal that this client is making on its own, I think they are in the _client_'s pending block, but not in the chain manager yet.
Maybe that's also the case if we're re-proposing something? Would be good to have that clearly defined.

Reply:
How would it manifest if client was making a proposal w/o some of the blobs being in its storageS yet? 🤔 If a client made a proposal and restarted before validators request blobs from it - would it be a liveness issue?

Reply:
Yes: That's why the blobs are treated as part of the pending block, which is persisted in the wallet. That way, a client can recover and finish finalizing its proposal even if it crashed or was closed.